applicationset/generators/matrix.go

```diff
@@ -168,6 +172,20 @@ func getMatrixGenerator(r argoprojiov1alpha1.ApplicationSetNestedGenerator) (*ar
 	return matrix.ToMatrixGenerator(), nil
 }
 
+func applyNestedMatrixSelectors(appSet *argoprojiov1alpha1.ApplicationSet, matrix *argoprojiov1alpha1.MatrixGenerator) {
```

Comment:
Can we consolidate this to one function and call it something like `dropDisabledSelectors`?

Reply:
Done, please check if the solution is okay like that.